src/main/kube-auth-proxy/get-kube-auth-proxy-certificate.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import type * as selfsigned from "selfsigned";
+
+type SelfSignedGenerate = typeof selfsigned.generate;
+
+const certCache: Map<string, selfsigned.SelfSignedCert> = new Map();
+
+export function getKubeAuthProxyCertificate(hostname: string, generate: SelfSignedGenerate, useCache = true): selfsigned.SelfSignedCert {
```

Comment:
`fromCache` might be better name since the cache is always used, in the sense that it is always updated (line 36) when generated (regardless of the value of `useCache`)